guides/javascript.md

```diff
@@ -0,0 +1,215 @@
+# JavaScript
+
+## Use the latest version of our ESLint config
+
+If you're working on a project, there's a good chance that you're using our shared ESLint rules, which codifies decisions we've made around code style and can be used to align that project with other projects in our ecosystem. Make sure the project is up to date with the various packages within the [`eslint-config`](https://github.com/MetaMask/eslint-config) monorepo.
+
+## Use a verb phrase for function/method names
+
+When naming a function or method, describe the action, not the outcome.
+
+🚫
+
+``` javascript
+function formattedChangelog() {
+  // ...
+}
+```
+
+✅
+
+``` javascript
+function formatChangelog() {
+  // ...
+}
+```
+
+For a function or method that returns a boolean, prefix the name with a descriptive verb so that readers don't confuse it with a boolean variable.
+
+🚫
+
+``` javascript
+class NetworkController {
+  isEIP1559Compatible() {
+    // ...
+  }
+}
+```
+
+✅
+
+``` javascript
+class NetworkController {
+  determineEIP1559Compatibility() {
```

Comment:
Hmm, not sure how I feel about this one. I like how concise `is[condition]` is.

I struggle to come up with non-is names for these cases too, particularly for the verb part. e.g. here it could be "determine" or 'calculate" or 'derive" or "check" or "verify" or "validate", etc. It's like a filler word.

Regarding the negative consequence:

>so that readers don't confuse it with a boolean variable

For me personally this is not a relatable experience. Maybe it's because of my editor highlighting, or because we tend to work more with TypeScript these days, but I can't recall ever confusing an `is___` function for a boolean variable.

Though... I do remember running into a similar problem. When we started introducing the ESLint `no-shadow` rule, I had to rename a bunch of `is___` variables to avoid conflict with `is___` functions.

Reply:
I couldn't find any examples of this guideline in the referenced style guides either

Reply:
> I do remember running into a similar problem. When we started introducing the ESLint `no-shadow` rule, I had to rename a bunch of `is___` variables to avoid conflict with `is___` functions.

Ah. I have run into this too. I wonder if that's a better rationale.

Reply:
I've made the requested change here: [6d93d07](https://github.com/MetaMask/contributor-docs/pull/11/commits/6d93d07187b2821d5d9de499bd10c77a714efb21)